File: Data Structures & Algorithms/maximum-number-of-balloons/submission-0.py
class Solution:
    def maxNumberOfBalloons(self, text: str) -> int:
        counter : List[int] = [0]*26
        for char in text :
            if char in ['a','b','n']: counter[ord(char)-ord('a')] +=1
            elif char in ['o','l'] : counter[ord(char)-ord('a')] +=1/2
            
        min : int = len(text) 

        for  char in 'balon':
            count : int =counter[ord(char)-ord('a')]
            if min >  count : min = count

            if min == 0 : return 0 

        return int(min) 

File: Data Structures & Algorithms/buy-and-sell-crypto/submission-0.java
class Solution {
    public int maxProfit(int[] prices) {


        int min = prices[0] , maxP = 0  ; 

        for(int i = 1 , n = prices.length ; i <n ;i++)
        {
            if(prices[i]- min > maxP) maxP =prices[i]-min ;
            if(min > prices[i]) min = prices[i];
        }

        return maxP ; 
        
    }
}
